src/test/main.cpp

```diff
@@ -24,3 +24,17 @@ const std::function<void(const std::string&)> G_TEST_LOG_FUN = [](const std::str
     if (!should_log) return;
     std::cout << s;
 };
+
+/**
+ * Retrieve the command line arguments from boost.
+ * Allows usage like:
+ * `test_bitcoin --run_test="net_tests/cnode_listen_port" -- -checkaddrman=1 -printtoconsole=1`
```

Comment:
There is an additional parameter `-cool` in the commit message of c79eb4ad9f0adf733a9da59d87dc3a0d10a90738, is this on purpose? Just noting that providing a non-recognized parameter like that causes an assertion for each unit test. I'm not sure if this is a problem or not.

Reply:
Well, the `-cool` is on purpose, like, not by accident, but may as well be removed. I wanted to use it to show that any other option is also ok there, like e.g `-putanyoptionyouwanthere`. Will remove it.

Reply:
Removed.

Reply:
> providing a non-recognized parameter like that causes an assertion for each unit test

Right, that is not very user (or developer) friendly. For example, the command:

```sh
./src/test/test_bitcoin --run_test="net_tests/cnode_listen_port" -- -foo
```

results in:

```
Assertion failed: (success), function BasicTestingSetup, file test/util/setup_common.cpp, line 104.
unknown location(0): fatal error: in "net_tests/cnode_listen_port": signal: SIGABRT (application abort requested)
```

changed to:

```
unknown location(0): fatal error: in "net_tests/cnode_listen_port": std::runtime_error: Invalid parameter -foo
```